### PR TITLE
feat(recordings): list virtualization

### PR DESCRIPTION
**Pull request by alexkim205**
## Changes

Screen Shot 2022-08-24 at 3 15 28 PM

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/list/PlayerList.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                                        const over = overscanCellsCount - under
+                                        return {
+                                            overscanStartIndex: Math.max(0, startIndex - under),
+                                            overscanStopIndex: Math.min(cellCount - 1, stopIndex + over),
+                                        }
+                                    }}
+                                    rowCount={data.length}
+                                    rowRenderer={({ index, style, key }) => {
+                                        const record = data[index] as T
+                                        const rowKeyDetermined = key ?? index
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/list/PlayerList.tsx:112]: nit: on first read, I'm not sure how to interpret the `Determimed` word here. Maybe a comment or change of name could help?

**Comment by rcmarron** [comment]
Just gave this a look through - looking good! I'm very curious to see how the underlying list library works when row heights are different (looks like it's supported), and if the row heights change - for example on row expand (I didn't see this in their example)

File: frontend/src/scenes/session-recordings/player/list/PlayerList.tsx (hunk 1 above)
Comment (alexkim205) [reply frontend/src/scenes/session-recordings/player/list/PlayerList.tsx:110]: I'm copying this terminology from LemonTable and TableRow, so I _think_ this should be fine.

**Comment by alexkim205** [comment]
Found a couple of bugs (fixing now):
- [x] console log jump to current time not working
- [x] dropdown menus for events and logs aren't populated

File: frontend/src/scenes/session-recordings/player/list/consoleLogsUtils.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        }
+    } catch {
+        /* Silently catch errors and continue to parse entry as string */
+    }
+
+    // Align all whitespace
+    rawEntry = rawEntry.replace(/\\n/g, ' ').replace(/\s+/g, ' ')
+
+    // Wrap urls with anchor tags
+    const boluses = rawEntry.split(' ').map((bolus) => {
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/list/consoleLogsUtils.tsx:41]: What does boluses mean?

File: frontend/src/scenes/session-recordings/player/list/PlayerListRow.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                        popup={{
+                            placement: 'bottom-end',
+                            overlay: (
+                                <>
+                                    <LemonButton fullWidth status="stealth">
+                                        Hello
+                                    </LemonButton>
+                                </>
+                            ),
+                        }}
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/list/PlayerListRow.tsx:115]: Should we just remove the elipsis until we build the features that go in there?

**Review by rcmarron — APPROVED** [review]
Just a few comments. I'm assuming we're planning to do the row expand stuff in a follow on PR

File: frontend/src/scenes/session-recordings/sessionRecordingLogic.ts [hunk 4]
```diff
@@ -249,7 +249,7 @@ export const sessionRecordingLogic = kea<sessionRecordingLogicType>([
                 eventUsageLogic
                     .findMounted()
                     ?.actions?.reportRecordingConsoleViewed(
-                        consoleLogsListLogic.findMounted()?.values?.consoleLogs?.length ?? 0
+                        consoleLogsListLogic.findMounted()?.values?.data?.length ?? 0
```
[the remark this answers is not in the document]
Comment (alexkim205) [reply frontend/src/scenes/session-recordings/sessionRecordingLogic.ts:252]: yeah good call ill move it to sharedlistlogic

File: frontend/src/scenes/session-recordings/player/list/consoleLogsUtils.tsx (hunk 2 above)
Comment (alexkim205) [reply frontend/src/scenes/session-recordings/player/list/consoleLogsUtils.tsx:41]: "a small rounded mass of a substance, especially of chewed food at the moment of swallowing" but i can see how thats not appropriate in software engineering lol

File: frontend/src/lib/components/LemonSelect.tsx [hunk 5]
```diff
@@ -7,7 +7,7 @@ import './LemonSelect.scss'
 import clsx from 'clsx'
 
 export interface LemonSelectOption<T> {
-    value: T
+    value?: T
```
Comment (Twixes) [inline frontend/src/lib/components/LemonSelect.tsx:10]: This made #11957 possible (see fix: #11958). It doesn't seem like `PlayerListRow`'s `options` are used anywhere yet – what should that menu look like? Maybe a `LemonMenu` component would be a better fit for it?

**Pull request merged**